overlord/servicestate/quota_control.go

```diff
@@ -123,14 +112,9 @@ func verifyQuotaRequirements(st *state.State, resourceLimits quota.Resources) er
 			return fmt.Errorf("cannot use journal quota with incompatible systemd: %v", err)
 		}
 
-		// We also need to check against experimental features for journal quota
-		if journErr := isQuotaAvailable(st, features.JournalQuota, "journal-quota"); journErr != nil {
-			// For backwards compatibility with current uses of journal quota before this added feature
-			// we also allow the enabled quota-groups feature switch to be set
-			if err := isQuotaAvailable(st, features.QuotaGroups, "quota-groups"); err != nil {
-				// but is that not enabled either, let's point them to the new option
-				return journErr
-			}
+		// To use journal quotas, the quota-groups experimental feature must be enabled.
+		if err := isQuotaAvailable(st, features.QuotaGroups, "quota-groups"); err != nil {
```

Comment:
as we are reusing the same experimental option, the API passing it in doesn't make a lot of sense, also the error probably needs to be more specific, as somebody might be setting many things and from the error as is it will be unclear what is triggering the problem currently